gulpfile.js

```diff
@@ -0,0 +1,306 @@
+ * @param {boolean} coverage Whether code coverage information should be generated.
+ */
+function makeTestFn(glob, coverage) {
+  const npmBin = (binName) => path.relative(outDir, `node_modules/.bin/${binName}`);
+  let result = () => {
+    let testNum = 0;
+    return gulp.src(glob)
+      .pipe(through2.obj((file, enc, cb) => {
+        const moduleAndArgs = [
+          ...coverage ? [
+            npmBin('istanbul'),
+            'cover',
+            '--dir',
+            `./coverage/${testNum++}`,
+            npmBin('_mocha'),
+            '--'
+          ] : [
+            npmBin('_mocha')
+          ],
+          '--require',
+          'source-map-support/register',
+          path.relative(outDir, file.path),
+          '--timeout',
+          '4000'
+        ];
+        const childProcess = cp.fork(
+          moduleAndArgs[0],
+          moduleAndArgs.slice(1),
+          { cwd: outDir }
+        );
+        childProcess.on('exit', (failures) => {
+          if (failures === 0) {
+            cb();
+          } else {
+            cb(new Error(`Mocha: ${failures} failures in ${file.path}]`));
+          }
+        })
+      }));
+  };
+  return result;
+}
+
+/**
+ * Creates a vinyl stream that decrypts a file.
+ */
+function decipher(key, iv) {
+  const decipher = crypto.createDecipheriv(
+    'aes-256-cbc',
+    Buffer.from(key, 'hex'),
+    Buffer.from(iv, 'hex')
+  );
+  return through2.obj(function (file, enc, cb) {
+    file.contents = Buffer.concat([decipher.update(file.contents, 'utf8'), decipher.final()]);
+    file.path = path.join(path.dirname(file.path), path.basename(file.path, '.enc'))
+    this.push(file);
+    cb();
+  });
+}
+
+/**
+ * Runs tslint on files in src/, with linting rules defined in tslint.json.
+ */
+gulp.task('lint', 'Emits linting errors found in src/ and test/.', () => {
+  if (util.env.dev) {
+    console.log('Skipping lint as the \'--dev\' option was passed.');
+    return merge2([]);
+  }
+  const program = require('tslint').Linter.createProgram(tsconfigPath);
+  return gulp.src(['src/**/*.ts', 'test/**/*.ts'])
+    .pipe(tslint({
+      configuration: tslintPath,
+      formatter: 'prose',
+      program
+    }))
+    .pipe(tslint.report({
+      summarizeFailureOutput: true
+    }))
+    .on('warning', onError);
+});
+
+gulp.task('format', 'Emits format errors found in src/ and test/.', () => {
+  if (util.env.dev) {
+    console.log('Skipping format as the \'--dev\' option was passed.');
+    return merge2([]);
+  }
+  throw new Error('Not implemented.');
+});
+
+gulp.task('clean', 'Deletes transpiled code.', () => {
+  return del(outDir);
+});
+
+/**
+ * Transpiles TypeScript files in src/ to JavaScript according to the settings found in tsconfig.json.
+ * Currently, all errors are emitted twice. This is being tracked here:
+ * https://github.com/ivogabe/gulp-typescript/issues/438
+ */
+gulp.task('compile', 'Transpiles src/.',
+  makeCompileFn({ transpile: ['*.ts', 'src/**/*.ts'] }));
```

Comment:
The globs are duplicated from tsconfig. Can we use `gts` for this?

Reply:
Done